firmware/eos.c

```diff
@@ -0,0 +1,476 @@
+#include "eos.h"
+
+#include "protect.h"
+#include "gettext.h"
+#include "layout2.h"
+#include "fsm.h"
+#include "util.h"
+#include "base58.h"
+#include "bip32.h"
+#include "hasher.h"
+#include "memzero.h"
+#include "secp256k1.h"
+
+#include "messages-eos.pb.h"
+
+#include <stdio.h>
+#include <time.h>
+
+Hasher hasher_preimage;
```

Comment:
In ours I marked these CONFIDENTIAL out of an abundance of caution. When I do that here, the section ends up too big.